tests/integration/graviton_test.py

```diff
@@ -773,7 +757,134 @@ def euclid(x, y):
         Invariant(predicate).validates(property, inputs, inspectors)
 
 
+def blackbox_pyteal_example4():
+    # Example 4: Using BlackboxPyTealer.program() to debug an ABIReturnSubroutine with an app, logic sig and csv report
+    from pathlib import Path
+    import random
+
+    from graviton.blackbox import DryRunExecutor, DryRunInspector
+
+    from pyteal import (
+        abi,
+        compileTeal,
+        ABIReturnSubroutine,
+        Expr,
+        For,
+        Int,
+        Mode,
+        ScratchVar,
+        Seq,
+        TealType,
+    )
+
+    from tests.blackbox import Blackbox, BlackboxPyTealer, algod_with_assertion
+
+    # Sum a dynamic uint64 array
+    @Blackbox(input_types=[None])
+    @ABIReturnSubroutine
+    def abi_sum(toSum: abi.DynamicArray[abi.Uint64], *, output: abi.Uint64) -> Expr:
+        i = ScratchVar(TealType.uint64)
+        valueAtIndex = abi.Uint64()
+        return Seq(
+            output.set(0),
+            For(
+                i.store(Int(0)),
+                i.load() < toSum.length(),
+                i.store(i.load() + Int(1)),
+            ).Do(
+                Seq(
+                    toSum[i.load()].store_into(valueAtIndex),
+                    output.set(output.get() + valueAtIndex.get()),
+                )
+            ),
+        )
+
+    # instantiate BlackboxPyTealer objects for the app and lsig:
+    app_pytealer = BlackboxPyTealer(abi_sum, Mode.Application)
+    lsig_pytealer = BlackboxPyTealer(abi_sum, Mode.Signature)
+
+    # create approval PyTeal app and lsig:
+    pyteal_abi_sum_app = app_pytealer.program()
+    pyteal_abi_sum_lsig = lsig_pytealer.program()
+
+    # compile the PyTeal's to Teal's:
+    teal_abi_sum_app = compileTeal(pyteal_abi_sum_app, Mode.Application, version=6)
+    teal_abi_sum_lsig = compileTeal(pyteal_abi_sum_lsig, Mode.Signature, version=6)
+
+    # infer the abi types for encoding/decoding dry runs:
+    abi_argument_types = app_pytealer.abi_argument_types()
+    abi_return_type = app_pytealer.abi_return_type()
+    # generate reports with the same random inputs (fix the randomness with a seed):
+    random.seed(42)
+
+    N = 50  # the number of dry runs for each experiment
+    choices = range(10_000)
+    inputs = []
+    for n in range(N):
+        inputs.append(tuple([random.sample(choices, n)]))
+
+    # execute the dry-run sequence:
+    algod = algod_with_assertion()
+
+    app_inspectors = DryRunExecutor.dryrun_app_on_sequence(
```

Comment:
@tzaffi Minor and optional:  As you have time, can you review https://github.com/algorand/pyteal/pull/335?

* https://github.com/algorand/pyteal/pull/335 illustrates a working example of an API change we discussed verbally that _may_ simplify blackbox test construction.
* I don't feel strongly about bringing in the change.  As noted in the PR, I see some drawbacks.  So, if you'd prefer as is, I'll happily close.

Reply:
I left some comments on #335